### PR TITLE
feat(web): add extension promo events

### PR DESCRIPTION
**Pull request by cbachmeier**
Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) and start with a valid [type](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type).

## Background
We want to track extension downloads originating from the web app
See this thread: https://uniswapteam.slack.com/archives/C047U65H422/p1720465268995329
...

## Changes
- Adds the following events:
  - UNISWAP_WALLET_EXTENSION_DOWNLOAD_OPENED
  - UNISWAP_WALLET_LEARN_MORE_CLICKED
  - UNISWAP_WALLET_PROMO_CLOSED
  - UNISWAP_WALLET_GETTING_STARTED_OPENED
- Adds the following elements:
  - EXTENSION_LAUNCH_WIDE_MODAL
  - EXTENSION_LAUNCH_DOWNLOAD_CONNECTOR
  - EXTENSION_LAUNCH_GETTING_STARTED

## Testing
Tested with new web promo banners

#### Before merging, please ensure the following:

- [x] All events have been approved by both product and engineering
- [x] All events have been tested in their consuming package
- [x] This PR is titled as `feat` for any new events, or otherwise follows conventional commit standards

### CODE REVIEW

File: src/interface/events.ts [hunk 1]
```diff
@@ -23,7 +23,11 @@ export enum InterfaceEventName {
   TOKEN_SELECTED = 'Token Selected',
   TOKEN_SELECTOR_OPENED = 'Token Selector Opened',
   UNISWAP_WALLET_APP_DOWNLOAD_OPENED = 'Uniswap Wallet App Download Opened',
+  UNISWAP_WALLET_EXTENSION_DOWNLOAD_OPENED = 'Uniswap Wallet Extension Download Opened',
+  UNISWAP_WALLET_GETTING_STARTED_OPENED = 'Uniswap Wallet Getting Started Modal Opened',
+  UNISWAP_WALLET_LEARN_MORE_CLICKED = 'Uniswap Wallet Learn More Clicked',
   UNISWAP_WALLET_MICROSITE_OPENED = 'Uniswap Wallet Microsite Opened',
+  UNISWAP_WALLET_PROMO_CLOSED = 'Uniswap Wallet Promotional Modal Closed',
```
Comment (adjkant) [inline src/interface/events.ts:30]: Can we do all of these as element names / page views with trace context? We shouldn't need separate events for all of these!

Comment (cbachmeier) [reply src/interface/events.ts:30]: these separate events were explicitly requested by product

here's the thread for more info https://uniswapteam.slack.com/archives/C047U65H422/p1720465268995329

Comment (cbachmeier) [reply src/interface/events.ts:30]: would the trace context catch if these aren't separate pages but instead modals found on different pages?

Comment (adjkant) [reply src/interface/events.ts:30]: Yes they can log with the specifics on different pages! We should log all steps but we don't need unique event names for these!

File: src/interface/trace.ts [hunk 2]
```diff
@@ -54,19 +54,25 @@ export enum InterfaceElementName {
...
   EXPLORE_POOLS_TAB = 'explore-pools-tab',
   EXPLORE_SEARCH_INPUT = 'explore_search_input',
   EXPLORE_TOKENS_TAB = 'explore-tokens-tab',
   EXPLORE_TRANSACTIONS_TAB = 'explore-transactions-tab',
+  EXTENSION_DOWNLOAD_BUTTON = 'extension-download-button',
+  EXTENSION_DOWNLOAD_CONNECTOR = 'extension-download-connector',
+  EXTENSION_LAUNCH_PROMOTIONAL_MODAL = 'extension-launch-promotional-modal',
   FIAT_ON_RAMP_BUY_BUTTON = 'fiat-on-ramp-buy-button',
   FIAT_ON_RAMP_LEARN_MORE_LINK = 'fiat-on-ramp-learn-more-link',
+  GETTING_STARTED_MODAL = 'GETTING_STARTED_MODAL',
```
Comment (adjkant) [inline src/interface/trace.ts:75]: these two can go to `InterfaceModalName`! Also matching formatting on `GETTING_STARTED_MODAL` to the rest!

**Review by adjkant — APPROVED** [review]

**Pull request merged**